### PR TITLE
feat: user module refactor part 1

### PR DESCRIPTION
**Pull request by xeno097**
### Changelog

This PR is the first of a series of pull requests aimed at refactoring the user module. The focus of this one is to refactor the infrastructure layer by hiding details implementation behind repository classes.
- Adds the `KycRequestRepository`, `KycRepository` and `UserRepository`.
- Adds the `KycRequestEntity`, `KycEntity` and `UserEntity` to split the core layer from the infra layer
- Moves the CognitoAuthProvider to the infra layer and implements the IAuthProvider base class
- Adds new errors fro handling user core flows

### CODE REVIEW

**Review by pellerano — APPROVED** [review]
Amazing work 🔥

File: src/modules/user/core/infra/auth-provider.ts [hunk 1]
```diff
@@ -179,8 +164,49 @@ export function authChallengeTypeFromStr(maybeType: string): AuthChallengeType {
...
+
+  const value = map[maybeCodeDeliveryMedium];
+
+  if (!value) {
+    throw new FailedStringToCodeliveryMediumConversion(maybeCodeDeliveryMedium);
   }
 
   return value;
 }
+
```
Comment (DHurtado714-itesm) [inline src/modules/user/core/infra/auth-provider.ts:195]: This is cool

File: src/modules/user/core/services/auth.service.new.ts [hunk 2]
```diff
@@ -206,65 +174,33 @@ export class NewAuthService {
...
-    return this.cognitoAuthProvider.forgotPassword(dto);
+    return this.authProvider.forgotPassword(dto);
   }
 
   public async completeForgotPassword(dto: CompleteForgotPasswordDto): Promise<void> {
-    return this.cognitoAuthProvider.completeForgotPassword(dto);
+    return this.authProvider.completeForgotPassword(dto);
   }
 
   public async deleteUserAccount(dto: DeleteUserAccountDto): Promise<void> {
```
Comment (DHurtado714-itesm) [inline src/modules/user/core/services/auth.service.new.ts:196]: Keep in mind there are transactions with the user id in the transactions table, we want to keep this records for metrics purpose

**Review by DHurtado714-itesm — APPROVED** [review]
LGTM

**Pull request merged**